apps/backend/src/lambdas/parseTDI.ts

```diff
@@ -0,0 +1,99 @@
+import { NestFactory } from '@nestjs/core';
+import { Context } from 'aws-lambda';
+import { AppModule } from '../app.module';
+import { AppLogger } from '../common/logger.service';
+import { S3ManagerService } from '../s3-manager/s3-manager.service';
+import { TDI17Record } from '../tdi-parser/tdi17/TDI17Record';
+import { TDI17Header } from '../tdi-parser/tdi17/TDI17Header';
+import { TDI17Details } from '../tdi-parser/tdi17/TDI17Details';
+import { TDI17Trailer } from '../tdi-parser/tdi17/TDI17Trailer';
+import { TDI34Record } from '../tdi-parser/tdi34/TDI34Record';
+import { TDI34Header } from '../tdi-parser/tdi34/TDI34Header';
+import { TDI34Details } from '../tdi-parser/tdi34/TDI34Details';
+import { TDI34Trailer } from '../tdi-parser/tdi34/TDI34Trailer';
+import { TDI34Enum } from './const/tdi34.const';
+import { TDI17Enum } from './const/tdi17.const';
+
+export const handler = async (
+  event?: { filePath: string; type: string },
+  context?: Context,
+) => {
+  const app = await NestFactory.createApplicationContext(AppModule);
+  const appLogger = app.get(AppLogger);
+  const s3manager = app.get(S3ManagerService);
+
+  appLogger.log({ event });
+  appLogger.log({ context });
+
+  const filetype = event?.type ?? process.env.TDI_TYPE;
```

Comment:
Suggestion - This line should throw an error if the event type does not exist. Defaulting should not be an option